lib/ASRouter.jsm

```diff
@@ -46,6 +81,50 @@ const MAX_MESSAGE_LIFETIME_CAP = 100;
 const LOCAL_MESSAGE_PROVIDERS = {OnboardingMessageProvider, CFRMessageProvider, SnippetsTestMessageProvider};
 const STARTPAGE_VERSION = "6";
 
+/**
+ * chooseBranch - choose from a list of optional branches and ratios
+ * @param branches {Array<[any, number?]>} A list of branches, where branch[0] is any item and branch[1] is the ratio
+ * @param seed {string} A unique seed for the randomizer
+ */
+async function chooseBranch(branches, seed) {
+  const ratios = branches.map(([item, ratio]) => ((typeof ratio !== "undefined") ? ratio : 1));
+  return branches[await Sampling.ratioSample(seed, ratios)][0];
```

Comment:
A `@return` comment above or here to be clear we're returning the "any" portion of the branch.